components/feature/privatemode/src/main/java/mozilla/components/feature/privatemode/notification/AbstractPrivateNotificationService.kt

```diff
@@ -69,15 +83,47 @@ abstract class AbstractPrivateNotificationService : Service() {
     @ExperimentalCoroutinesApi
     final override fun onCreate() {
         val id = SharedIdsHelper.getIdForTag(this, NOTIFICATION_TAG)
-        val channelId = ensureNotificationChannelExists(this, NOTIFICATION_CHANNEL, onSetupChannel = {
-            if (SDK_INT >= Build.VERSION_CODES.O) {
-                enableLights(false)
-                enableVibration(false)
-                setShowBadge(false)
-            }
-        })
-
-        val notification = NotificationCompat.Builder(this, channelId)
+        val channelId =
+            ensureNotificationChannelExists(
+                this,
+                NOTIFICATION_CHANNEL,
+                onSetupChannel = {
+                    if (SDK_INT >= Build.VERSION_CODES.O) {
+                        enableLights(false)
+                        enableVibration(false)
+                        setShowBadge(false)
+                    }
+                }
+            )
+
+        val notification = createNotification(channelId)
+
+        startForeground(id, notification)
+
+        privateTabsScope = store.flowScoped { flow ->
+            flow.map { state -> state.privateTabs.isEmpty() }
+                .ifChanged()
+                .collect { noPrivateTabs ->
+                    if (noPrivateTabs) stopService()
+                }
+        }
+
+        localeScope = store.flowScoped { flow ->
+            flow.mapNotNull { state -> state.locale }
+                .ifChanged()
+                .collect {
+                    notifyLocaleChanged(id, channelId)
```

Comment:
I thought this is where we should be updating the notification?  Since the notification is created here in the abstract service side?

Reply:
We want Fenix (or whoever uses this component) to be able to define what they want to do when the locale has changed

Reply:
Yeah, you could see the implementation on the Fenix [pr](https://github.com/mozilla-mobile/fenix/pull/18179/files) [specifically](https://github.com/mozilla-mobile/fenix/blob/9dc713b363521ad243bdfd551f5875c97e4e575e/app/src/main/java/org/mozilla/fenix/session/PrivateNotificationService.kt#L61)

Reply:
Per our conversation @rocketsroger, I created a function inside the service that does the notifying. That way, the consumer can still decide what it wants to do, and we are providing more functionality to the user by having the build & notify in the service itself to be used when needed